src/lwc/rd2EntryForm/rd2EntryForm.js

```diff
@@ -413,87 +392,142 @@ export default class rd2EntryForm extends LightningElement {
-                    allFields[FIELD_COMMITMENT_ID.fieldApiName] = tempId;
+                if (response.statusCode === HTTP_CODES.Created
+                    || response.statusCode === HTTP_CODES.OK
+                ) {
+                    this.populateCommitmentFields(response, allFields);
+
+                    this.processSubmit(allFields);
+
+                } else {
+                    let message = this.getCommitmentError(response);
+                    this.handleSaveError(message);
                 }
+            })
+            .catch((error) => {
+                this.handleSaveError(error);
+            });
+    }
 
-            } catch (error) {
-                this.setSaveButtonDisabled(false);
-                return;
-            }
+    /***
+    * @description Determines if the Recurring Donation is an Elevate recurring commitment
+    */
+    isCommitment() {
+        if (!this.isElevateCustomer) {
+            return false;
         }
 
-        try {
-            this.loadingText = this.customLabels.savingRDMessage;
-            this.template.querySelector('[data-id="outerRecordEditForm"]').submit(allFields);
+        if (this.isEdit) {
+            return !isNull(this.commitmentId);
 
-        } catch (error) {
-            this.handleSaveError(error);
+        } else {
+            // A new Recurring Donation will be a new Elevate recurring commitment
+            // when the Elevate widget is displayed on the entry form.
+            return this.isElevateWidgetDisplayed();
         }
     }
 
-    /***
-    * @description Collects fields displayed on the form and any integrated LWC
-    */
-    getAllSectionsInputValues() {
-        const donorFields = (isNull(this.donorComponent))
-            ? {}
-            : this.donorComponent.returnValues();
+    constructRecurringDonation(allFields) {
+        let rd = { ...allFields };
 
-        const scheduleFields = (isNull(this.scheduleComponent))
-            ? {}
-            : this.scheduleComponent.returnValues();
+        const installmentFrequency = rd[FIELD_INSTALLMENT_FREQUENCY.fieldApiName];
+        if (isNull(installmentFrequency) || isUndefined(installmentFrequency)) {
+            rd[FIELD_INSTALLMENT_FREQUENCY.fieldApiName] = 1;
+        }
 
-        const extrafields = (isNull(this.customFieldsComponent))
-            ? {}
-            : this.customFieldsComponent.returnValues();
+        if (this.commitmentId) {
+            rd[FIELD_COMMITMENT_ID.fieldApiName] = this.commitmentId;
+        }
 
-        return { ...scheduleFields, ...donorFields, ...extrafields, ...this.returnValues() };
+        return rd;
     }
 
-    /***
-    * @description Validate all fields on the integrated LWC sections
-    */
-    isSectionInputsValid() {
-        const isDonorSectionValid = (isNull(this.donorComponent))
-            ? true
-            : this.donorComponent.isValid();
+    populateCommitmentFields(response, allFields) {
```

Comment:
After creating or editing the commitment in Elevate, bring back relevant fields into the RD record (W-8554656 and W-8585272).